### PR TITLE
Quick attribute filter for feature grid 

### LINKED ISSUE #2189: Quick search filter for feature grid
We should add to the feature grid a way to search quickly on attributes content. 

We should use this extension to allow the user to quickly search for a string, a number or a date/time. 

http://adazzle.github.io/react-data-grid/examples.html#/custom-filters
Notes:
 - You should be able to disable this functionality by configuration
 - This functionality must be locked in edit mode
 - Quick search should be a valid alternative to advanced search in general
 - Let's keep the possibility make quick filter and advanced filter synchronized, for the future (now they are not synchronized, an advanced search query will reset the quick search).

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
Fix #2189. Provide a support to quick search on string, number, date ... attributes value. 

## Closed issues

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")
 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
You must filter using the advanced search

**What is the new behavior?**
You can perform quick search directly in the feature grid

**Does this PR introduce a breaking change?** (check one with "x")
 - [x] No

### CODE REVIEW

File: web/client/components/data/featuregrid/filterRenderers/AttributeFilter.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            return <span/>;
+        }
+        const placeholder = LocaleUtils.getMessageById(this.context.messages, this.props.placeholderMsgId) || "Search";
+        let inputKey = 'header-filter-' + this.props.column.key;
+        return (<input disabled={this.props.disabled} key={inputKey} type="text" className="form-control input-sm" placeholder={placeholder} value={this.props.value} onChange={this.handleChange}/>);
+    }
+    renderTooltip = (cmp) => {
+        if (this.props.tooltipId && LocaleUtils.getMessageById(this.context.messages, this.props.tooltipId)) {
+            return (<OverlayTrigger placement="top" overlay={<Tooltip id="tooltip">{LocaleUtils.getMessageById(this.context.messages, this.props.tooltipId)}</Tooltip>}>
+              {cmp}
```
Comment (MV88) [inline web/client/components/data/featuregrid/filterRenderers/AttributeFilter.jsx]: you shouldn't use the tooltipid as id for the tooltip component? 
placement is not customizable

File: web/client/selectors/featuregrid.js [hunk 2]
```diff
@@ -46,9 +46,13 @@ const hasGeometrySelectedFeature = (state) => {
 };
 const hasChangesSelector = state => changesSelector(state) && changesSelector(state).length > 0;
 const hasNewFeaturesSelector = state => newFeaturesSelector(state) && newFeaturesSelector(state).length > 0;
+const getAttributeFilters = state => state && state.featuregrid && state.featuregrid.filters;
```
Comment (MV88) [inline web/client/selectors/featuregrid.js]: no use of get here?

File: web/client/translations/data.es-ES [hunk 3]
```diff
@@ -876,6 +876,19 @@
             "featureClose": "Realmente desea cerrar la parrilla?",
             "delete": "Confirmar para eliminar los elementos?",
             "missingGeometry": "Missing geometry",
+            "filter": {
+                "placeholders": {
+                    "default": "Buscar...",
+                    "string": "Escribe texto para filtrar...",
+                    "number": "Escriba el número o la expresión...."
+                },
```
Comment (MV88) [inline web/client/translations/data.es-ES]: 4 points here

File: web/client/selectors/featuregrid.js (hunk 2 above)
Comment (offtherailz) [reply web/client/selectors/featuregrid.js]: If it does not change, this version is more efficient

**Pull request merged**